ext/afform/core/Civi/Afform/PageTokenCredential.php

```diff
@@ -31,19 +32,31 @@ public static function getSubscribedEvents(): array {
    * If you visit a top-level page like "civicrm/my-custom-form?_aff=XXX", then
    * all embedded AJAX calls should "_authx=XXX".
    *
-   * @param array $path
+   * @param \Civi\Core\Event\GenericHookEvent $e
    * @return void
    */
-  public function onInvoke(array $path) {
+  public function onInvoke(GenericHookEvent $e) {
     $token = $_REQUEST['_aff'] ?? NULL;
 
     if (empty($token)) {
       return;
     }
+
     if (!preg_match(';^[a-zA-Z0-9\.\-_ ]+$;', $token)) {
       throw new \CRM_Core_Exception("Malformed page token");
     }
 
+    // FIXME: This would authenticate requests to the main page, but it also has the side-effect
+    // of making the user login.
+
+    // \CRM_Core_Session::useFakeSession();
+    // $params = ($_SERVER['REQUEST_METHOD'] === 'GET') ? $_GET : $_POST;
+    // $authenticated = \Civi::service('authx.authenticator')->auth($e, ['flow' => 'param', 'cred' => $params['_aff'], 'siteKey' => NULL]);
```

Comment:
can't you pass `useSession` => `FALSE` here to make it stateless for the top level page load?

Reply:
Yeah, you would think so. I'm pretty sure I dipped a toe in that direction -- setting the explicit value and/or inspecting the effective value (with debugger). But while inspecting I realized that `useSession=>FALSE` is [already the default](https://github.com/civicrm/civicrm-core/blob/92a32996710ed8f19ac9ff30e38c566a05bd9ac2/ext/authx/Civi/Authx/Authenticator.php#L127).

It's particularly strange because `StatelessFlowsTest` is [passing](https://test.civicrm.org/job/CiviCRM-Test/23459/testReport/(root)/Civi_Authx_StatelessFlowsTest/). (*If the problem was on J/WP/SA, then [this known quirk](https://github.com/civicrm/civicrm-core/blob/7d8c7388292314ef99bae6ed1cb79453b3982cb0/ext/authx/tests/phpunit/Civi/Authx/AbstractFlowsTest.php#L380-L382) could explain the discrepancy. But I'm pretty sure my final testing was on D7 which should be more robust.*)

Of course, it is worth double-checking the simple things again. 

But this is why I'm expecting it needs a deeper trace of the session-management. (*...Like maybe the `FakeSession` isn't superceding the UF's default session for some reason? Or something in UF-land is forcing the session to start?...*)